File: livingdoc-api/src/main/java/org/livingdoc/api/fixtures/decisiontables/AfterRow.java
package org.livingdoc.api.fixtures.decisiontables;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Methods annotated with this annotation are invoked once for each row of the decision table example just after it was
 * evaluated. These methods are intended to close resources initialized in {@link BeforeRow} annotated methods.
 * <p>
 * <b>Constraints:</b>
 * <ol>
 * <li>The annotated method must not have any parameters!</li>
 * <li>If multiple methods of a single fixture are annotated the invocation order is non-deterministic!</li>
 * </ol>
 *
 * @see DecisionTableFixture
 * @since 2.0
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface AfterRow {
}

File: livingdoc-api/src/main/kotlin/org/livingdoc/api/fixtures/decisiontables/AfterRow.kt
package org.livingdoc.api.fixtures.decisiontables

/**
 * Methods annotated with this annotation are invoked once for each row of the decision table example just after it was
 * evaluated. These methods are intended to close resources initialized in [BeforeRow] annotated methods.
 *
 * **Constraints:**
 *  1. The annotated method must not have any parameters!
 *  1. If multiple methods of a single fixture are annotated the invocation order is non-deterministic!
 *
 * @see DecisionTableFixture
 *
 * @since 2.0
 */
@Target(AnnotationTarget.FUNCTION)
annotation class AfterRow
